spring-xd-dirt/src/main/java/org/springframework/xd/dirt/plugins/spark/streaming/SparkStreamingPlugin.java

```diff
@@ -73,19 +74,27 @@ public void postProcessModule(Module module) {
 		Properties messageBusProperties = getMessageBusProperties(module, transport);
 		Properties inboundModuleProperties = this.extractConsumerProducerProperties(module)[0];
 		Properties outboundModuleProperties = this.extractConsumerProducerProperties(module)[1];
-		String defaultStorageLevel = env.getProperty(Processor.SPARK_STORAGE_LEVEL_PROP);
+		String defaultStorageLevel = env.getProperty(SparkStreamingSupport.SPARK_STORAGE_LEVEL_PROP);
 		StorageLevel configuredStorageLevel = StorageLevel.fromString(StringUtils.hasText(defaultStorageLevel) ?
 				defaultStorageLevel : DEFAULT_STORAGE_LEVEL);
-		String storageLevelFromModule = module.getProperties().getProperty(Processor.SPARK_STORAGE_LEVEL_MODULE_OPTION);
+		String storageLevelFromModule = module.getProperties().getProperty(SparkStreamingSupport.SPARK_STORAGE_LEVEL_MODULE_OPTION);
 		StorageLevel storageLevel = StringUtils.hasText(storageLevelFromModule) ?
 				StorageLevel.fromString(storageLevelFromModule) : configuredStorageLevel;
 		MessageBusReceiver receiver = null;
 		if (transport.equals("local")) {
-			final Processor sparkModule = module.getComponent(Processor.class);
-			Properties sparkConfigs = SparkStreamingDriverModule.getSparkModuleProperties(sparkModule);
-			String sparkMasterUrl = env.getProperty(Processor.SPARK_MASTER_URL_PROP);
-			if (sparkConfigs != null && StringUtils.hasText(sparkConfigs.getProperty(Processor.SPARK_MASTER_URL_PROP))) {
-				sparkMasterUrl = sparkConfigs.getProperty(Processor.SPARK_MASTER_URL_PROP);
+			SparkStreamingSupport processor;
+			Properties sparkConfigs = null;
+			try {
+				processor = module.getComponent(SparkStreamingSupport.class);
+				Assert.notNull(processor, "Problem getting the spark streaming module. Is the module context active?");
+				sparkConfigs = SparkStreamingDriverModule.getSparkModuleProperties(processor);
+			}
+			catch (NoSuchBeanDefinitionException e) {
```

Comment:
This should probably throw something more specific than RuntimeException, perhaps IllegalStateException.